### PR TITLE
networkUp only after service active for 5s

### PR DESCRIPTION
**Pull request by kinarashah**
[rancher/rancher#10217](https://github.com/rancher/rancher/issues/10217)

### CODE REVIEW

File: code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java [hunk 1]
```diff
@@ -22,15 +24,20 @@
...
 import javax.inject.Inject;
 import javax.inject.Named;
 
 
+
 @Named
 public class ServiceEventCreate extends AbstractObjectProcessHandler implements ProcessHandler, Priority {
 
     private static final List<String> UP_STATES = Arrays.asList(CommonStatesConstants.ACTIVE, CommonStatesConstants.UPDATING_ACTIVE);
+    private static final Long SECOND_MILLIS = 1000L;
```
Comment (alena1108) [inline code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java]: I'd just make configuration in miliseconds. No real need to define 1000L

File: code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java [hunk 2]
```diff
@@ -104,6 +111,19 @@ private boolean isNetworkUp(long accountId) {
                 return false;
             }
         }
+        if(networkDriverService.getState().equals(CommonStatesConstants.ACTIVE)) {
+            Object obj = DataAccessor.fields(networkDriverService).withKey(ServiceConstants.LAST_ACTIVE).get();
```
Comment (alena1108) [inline code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java]: there is a DataAccessor.FieldDate method that can be used

File: code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java [hunk 3]
```diff
@@ -22,15 +24,20 @@
...
 import javax.inject.Named;
 
 
+
 @Named
 public class ServiceEventCreate extends AbstractObjectProcessHandler implements ProcessHandler, Priority {
 
     private static final List<String> UP_STATES = Arrays.asList(CommonStatesConstants.ACTIVE, CommonStatesConstants.UPDATING_ACTIVE);
+    private static final Long SECOND_MILLIS = 1000L;
+    private static final long STAY_ACTIVE_SECONDS = ArchaiusUtil.getLong("stay_active.seconds").getValue() * SECOND_MILLIS;
```
Comment (alena1108) [inline code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java]: Should we name it with correlation with service.lastActive parameter?

File: code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java (hunk 2 above)
Comment (kinarashah) [reply code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java]: It internally converts the long value to Date and returns null if key not present, I'll need to check for null and convert back to long anyway. 
Or am I understanding it wrong ?

File: code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java (hunk 3 above)
Comment (kinarashah) [reply code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/process/ServiceEventCreate.java]: lastActiveTime = time service was last active 
Changing it to, STAY_ACTIVE_MILLISECONDS = duration network service needs to stay active for before network's considered Up. 
Does that sound ok?

**Comment by alena1108** [comment]
@kinarashah can you look at the failure to see if it is related to the changes we've made? Looks like it might.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**